sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileSmbProperties.java

```diff
@@ -0,0 +1,278 @@
+     * @param fileCreationTime The file creation time.
+     * @return the updated FileSmbProperties object..
+     */
+    public FileSmbProperties fileCreationTime(OffsetDateTime fileCreationTime) {
+        this.fileCreationTime = fileCreationTime;
+        return this;
+    }
+
+    /**
+     * Sets the file last write time.
+     * @param fileLastWriteTime The file last write time.
+     * @return the updated FileSmbProperties object.
+     */
+    public FileSmbProperties fileLastWriteTime(OffsetDateTime fileLastWriteTime) {
+        this.fileLastWriteTime = fileLastWriteTime;
+        return this;
+    }
+
+    // HELPER METHODS
+
+    /**
+     * Determines the value of the file permission header.
+     * @param filePermission The file permission.
+     * @param defaultValue The default file permission header value.
+     * @return The value of the file permission header
+     */
+    String filePermission(String filePermission, String defaultValue) {
+        return (filePermission == null) && (filePermissionKey == null) ?
+            defaultValue :
+            filePermission;
+    }
+
+    /**
+     * Determines the value of the ntfs attributes header.
+     * @param defaultValue The default ntfs attributes header value.
+     * @return The value of the ntfs attributes header
+     */
+    String ntfsFileAttributes(String defaultValue) {
+        return ntfsFileAttributes == null ?
+            defaultValue :
+            NtfsFileAttributes.toString(ntfsFileAttributes);
+    }
+
+    /**
+     * Determines the value of the creation time header.
+     * @param defaultValue The default creation time header value.
+     * @return The value of the creation time header
+     */
+    String fileCreationTime(String defaultValue) {
+        return fileCreationTime == null ?
+            defaultValue :
+            parseFileSMBDate(fileCreationTime);
+    }
+
+    /**
+     * Determines the value of the last write time header.
+     * @param defaultValue The default last write time header value.
+     * @return The value of the last write time header
+     */
+    String fileLastWriteTime(String defaultValue) {
+        return fileLastWriteTime == null ?
+            defaultValue :
+            parseFileSMBDate(fileLastWriteTime);
+    }
+
+    /**
+     * Given an <code>OffsetDateTime</code>, generates a {@code String} representing a date in the format needed for
+     * file SMB properties
+     * @param time the <code>OffsetDateTime</code> to be interpreted as a {@code String}
+     * @return The {@code String} representing the date
+     */
+    private static String parseFileSMBDate(OffsetDateTime time) {
+        return time.format(DateTimeFormatter.ofPattern(FileConstants.SMB_DATE_STRING));
+    }
+
+    /**
+     * Creates a new {@link FileSmbProperties} from a {@link FilesCreateResponse}.
+     * @param response The {@link FilesCreateResponse}.
+     */
+    FileSmbProperties(FilesCreateResponse response) {
```

Comment:
For all of these creation methods are the same properties accessed with the same names? It might be worse but a thought could be taking a HttpHeaders object and pulling the values out manually, that way there would only be one method to maintain.

Reply:
This is a good idea, I might first finish up the get and create share since its an important part of this feature then take some time to refactor this